### ISSUE TITLE
How to set parameters of SfMPerspectiveCamera

### ISSUE THREAD

**Issue by yuhaoooo** [facebookresearch/pytorch3d#272]
## ❓ How to set parameters of SfMPerspectiveCamera

I have a problem when calculating the intrinsic matrix K. I have a 3D points(world coordinates) and the corresponding 2D points(screen space). And also, I have the extrinsic matrix (R and T), which is predicted by a deep network. So I use the `least sqaures algorithm` to estimate K, which is as follows:
```
def POS(xp, x):
    # calculating least sqaures problem
    # ref https://github.com/pytorch/pytorch/issues/27036
    ls = LeastSquares()
    npts = xp.shape[1]

    A = torch.zeros([2*npts, 4]).to(x.device)
    A[0:2*npts-1:2, 0:2] = x[0, :, [0, 2]]
    A[1:2*npts:2, 2:4] = x[0, :, [1, 2]]

    b = torch.reshape(xp[0], [2*npts, 1])

    k = ls.lstq(A, b, 0.010)

    fx = k[0, 0]
    px = k[1, 0]
    fy = k[2, 0]
    py = k[3, 0]
    return fx, px, fy, py
```

Meanwhile, I noticed the issue #184, #171, and convert the `fx, px, fy, py` in screen space to NDC space.
```
half_size = image_size / 2
focal_length = torch.tensor([fx / half_size, fy / half_size], dtype=torch.float32).reshape(1, 2).to(device)
principal_point = torch.tensor([(px - half_size) / half_size, (py - half_size) / half_size], dtype=torch.float32).reshape(1, 2).to(device)

cameras = SfMPerspectiveCameras(
        device=device,
        R=R,
        T=T,
        focal_length=focal_length,
        principal_point=principal_point
)
```

However,  when I use the following code to check the `K`, the results is not what I think.
```
trans = cameras.get_full_projection_transform()
landmarks_2d = trans.transform_points(landmarks_3d)
```

Is there something wrong?

**Comment by gkioxari** [comment]
Hi @yuhaoooo 
This question is not related to PyTorch3D and we will not addressing issues that are not bugs. Unfortunately, we can't be consultants to projects done by users and help them out, otherwise that's all we would be doing. 

My advice to you is to take a pen and a paper and figure out the math of what you want to do and how you want to transform your points. Indeed, our cameras as they are written now, when they interface with the renderer, expect the parameters to be in NDC space and not in screen space. The transform between the two spaces is really easy. For x-dimension, NDC is from [-1, 1] and maps to screen which is from [0, W-1]. For y-axis, NDC is from [-1, 1] and maps to screen which is from [0, H-1]. The transform I provided in the issues above does exactly this mapping. I hope this advice was  useful in debugging your use case.

**Comment by yuhaoooo** [comment]
Thanks a lot for your responds. :)
